src/careamics/config/algorithm_model.py

```diff
@@ -134,21 +134,6 @@ def algorithm_cross_validation(self: Self) -> Self:
                     "sure that `in_channels` and `num_classes` are the same."
                 )
 
-        # N2N
-        if self.algorithm == "n2n":
-            # n2n is only compatible with the UNet model
-            if not isinstance(self.model, UNetModel):
-                raise ValueError(
-                    f"Model for algorithm {self.algorithm} must be a `UNetModel`."
-                )
```

Comment:
Yes!

The thought is that (as far as I thought) there is no reason to make N2N not compatible with the future VAE.

Reply:
I see. In the convenience function `create_n2n_configuration`, that in turn calls `_create_supervised_configuration` a UNet is initialised (see code ref below). When the option of different models are fully implemented should, it be possible for the model to be selected by the user through the convenience function?

https://github.com/CAREamics/careamics/blob/8c344920387d964c28839c53152d286ad53af9fa/src/careamics/config/configuration_factory.py#L87-L98

Reply:
I would be in favor of keeping the convenience function like this while the VAE are not available, rather than having a parameter that will throw a `NotImplementedError` (which would be thrown if people try to use a VAE in a self crafter configuration).

Once the VAE is available, we would indeed need to reflect this in the convenience functions. (which means we would need to remember that, but worst case it comes up as an issue at some point).

Does that make sense?

Reply:
yes, I meant change it only once other options are available